src/loadRc.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+'use strict';
+
+const yaml = require('js-yaml');
+const requireFromString = require('require-from-string');
+const readFile = require('./readFile');
+const parseJson = require('./parseJson');
+const funcRunner = require('./funcRunner');
+
+module.exports = function loadRc(
+  filepath: string,
+  options: {
+    sync?: boolean,
+    rcStrictJson?: boolean,
+    rcExtensions?: boolean,
+  }
+): Promise<?cosmiconfig$Result> | ?cosmiconfig$Result {
+  if (!options.sync) {
+    return readFile(filepath)
+      .then(parseExtensionlessRcFile)
+      .then(checkExtensionlessRcResult);
+  } else {
+    return checkExtensionlessRcResult(
+      parseExtensionlessRcFile(readFile.sync(filepath))
+    );
+  }
+
+  function checkExtensionlessRcResult(result) {
+    if (result) return result;
+    if (options.rcExtensions) return loadRcWithExtensions();
+    return null;
+  }
+
+  function parseExtensionlessRcFile(content: ?string): ?cosmiconfig$Result {
+    if (!content) return null;
+    const pasedConfig = options.rcStrictJson
+      ? parseJson(content, filepath)
+      : yaml.safeLoad(content, { filename: filepath });
+    return {
+      config: pasedConfig,
+      filepath,
+    };
+  }
+
+  function loadRcWithExtensions() {
+    let foundConfig = null;
```

Comment:
Is there anyway we can avoid this? I am uncomfortable with the idea of anonymous functions modifying this variable. If adding flow is complicating it, can we skip it for this file?

Reply:
I prefer the clarity now, with or without Flow. 

> I am uncomfortable with the idea of anonymous functions modifying this variable.

Why is that? 

It seemed to me the alternative is to make each function a little more ambiguous: might receive a string or a Result object or a falsy value. Having the condition that duck types the argument to check if it is a Result object or a string didn't feel any cleaner to me.

Reply:
Okay. I'll try to refactor this sometime later using [`p-if`](https://github.com/sindresorhus/p-if)

Reply:
It sounds to me like that might be unnecessary work that will add new complexity and abstractions that make the code harder to read. But if you want to give it a shot, happy to look.

Reply:
I'll try to refactor in a way which is easy to understand. If I am not able to do that, will leave as is.